services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DeviceDto.java

```diff
@@ -91,8 +89,10 @@ public static DeviceDto forRead(final String tenantId, final String deviceId, fi
         final DeviceDto deviceDto = BaseDto.forRead(DeviceDto::new, device, created, updated, version);
         deviceDto.setTenantId(tenantId);
         deviceDto.setDeviceId(deviceId);
-        deviceDto.setIsAutoProvisioned(autoProvisioned);
-        deviceDto.setIsAutoProvisioningNotificationSent(autoProvisioningNotificationSent);
+        deviceDto.setDeviceStatus(new DeviceStatus()
```

Comment:
is there a particular reason why we do not simply pass a *DeviceStatus* instance as a parameter to this method? Or do you want to hide the existence of *DeviceStatus* to client code?

Reply:
No, I hadn't information hiding in mind when I implemented this. Thinking a second time about this, your proposal makes sense as it shortens the parameter list.